packages/@lwc/engine-core/src/framework/vm.ts

```diff
@@ -256,10 +255,6 @@ function resetComponentStateWhenRemoved(vm: VM) {
         runChildNodesDisconnectedCallback(vm);
         runLightChildNodesDisconnectedCallback(vm);
     }
-
-    if (process.env.NODE_ENV !== 'production') {
-        removeActiveVM(vm);
-    }
```

Comment:
`removeActiveVM` is no longer needed because we're just relying on the browser's built-in GC for cleanup.